### PR TITLE
Botany

### PR DESCRIPTION
**Pull request by Zumorica**
- [x] Seeds
- [x] Trays
- [x] Planting
- [x] Harvesting
- [x] Plants die
- [x] Plants need nutriment and water
- [x] Pests and weeds
- [x] Plants consume/exude gasses
- [x] Plants need ideal temperatures
- [x] Produce has different amounts of different reagents depending on the seed
- [x] Tray lights
- [x] Botany tools and sampling
- [x] ????
- [x] the rest of the fucking owl

Future PRs:
- Seed extractor UI
- Mapping botany into saltern
- Plant mutations
- More plants
- More plant qualities!
- Making tomatoes go *splat*
- Closed system hydroponics
- Mysterious seeds that appear in maint!
- Light levels. *shiver*

### CODE REVIEW

**Comment by Zumorica** [comment]
Ready for review.

**Comment by Aranclanos** [comment]
tomato plants disappear after the first harvest
sugarcanes dont disappear after the first harvest but honestly I have no idea if they should
water/nutrientment values go down extremely slowly, see image

image

**Comment by Zumorica** [comment]
> tomato plants disappear after the first harvest

fixed now, thanks for pointing it out

> sugarcanes dont disappear after the first harvest but honestly I have no idea if they should

porting botany from VG, they don't seem to repeat harvest there, I think?

> water/nutrientment values go down extremely slowly, see image

I'll mess with the values and come back to you

**Comment by PJB3005** [comment]
> porting botany from VG, they don't seem to repeat harvest there, I think?

They do:

```dm
/datum/seed/sugarcane
	name = "sugarcane"
	seed_name = "sugarcane"
	display_name = "sugarcanes"
	plant_dmi = 'icons/obj/hydroponics/sugarcane.dmi'
	products = list(/obj/item/weapon/reagent_containers/food/snacks/grown/sugarcane)
	harvest_repeat = 1
	chems = list(SUGAR = list(4,5))

	lifespan = 60
	maturation = 3
	production = 6
	yield = 4
	potency = 10
	growth_stages = 3
	ideal_heat = 298
```

**Comment by Zumorica** [comment]
Welp I totally missed that harvest_repeat = 1

**Comment by Zumorica** [comment]
Looking at it, I already set sugarcanes to repeat harvest. Might be a look so I'll look into it

**Comment by Zumorica** [comment]
@Aranclanos hey, I balanced consumption so it consumes 2 times more likely, and 4 times as much. Feel free to test and tell me how it feels now! Also I'm pretty sure sugarcanes repeat harvest juts fine hmm

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs [hunk 1]
```diff
@@ -0,0 +1,850 @@
...
+                Seed.SpawnSeedPacket(Owner.Transform.Coordinates);
+                RemovePlant();
+                ForceUpdate = true;
+                Update();
+            }
+
+            CheckHealth();
+
+            if (Harvest && Seed.HarvestRepeat == HarvestType.SelfHarvest)
+                AutoHarvest();
```
Comment (brndd) [inline Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:379]: Shouldn't this be below the harvest check below or do you intentionally want to delay autoharvest to the next Update tick?

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs [hunk 2]
```diff
@@ -0,0 +1,850 @@
...
+                    sprayed = true;
+                    amount = 1f;
+                    EntitySystem.Get<AudioSystem>().PlayFromEntity(spray.SpraySound, usingItem, AudioHelpers.WithVariation(0.125f));
+                }
+
+                var chemAmount = ReagentUnit.New(amount);
+
+                var split = solution.Solution.SplitSolution(chemAmount <= solution.Solution.TotalVolume ? chemAmount : solution.Solution.TotalVolume);
+
+                user.PopupMessageCursor(sprayed ? $"You spray {Owner.Name} with {usingItem.Name}." : $"You transfer {split.TotalVolume.ToString()}u to {Owner.Name}");
```
Comment (brndd) [inline Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:715]: not localized

File: Content.Shared/Atmos/Atmospherics.cs [hunk 3]
```diff
@@ -231,7 +231,7 @@ public class Atmospherics
     }
 
     /// <summary>
-    ///     Gases to Ids. Keep these updated with the prototypes!
+    ///     Gasses to Ids. Keep these updated with the prototypes!
```
Comment (brndd) [inline Content.Shared/Atmos/Atmospherics.cs:234]: The plural is in fact "gases"

File: Content.Shared/Interfaces/Chemistry/IPlantMetabolizable.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Content.Shared.Interfaces.Chemistry
+{
+    public interface IPlantMetabolizable : IExposeData
+    {
+        /// <summary>
+        ///     Metabolize 1 unit of a reagent.
+        /// </summary>
+        /// <param name="plantHolder"></param>
+        /// <param name="reactVolume"></param>
```
Comment (brndd) [inline Content.Shared/Interfaces/Chemistry/IPlantMetabolizable.cs:14]: seems like it should actually be `Metabolize <paramref name="reactVolume"/> units of a reagent.`

File: Resources/Prototypes/Catalog/VendingMachines/seeds.yml [hunk 5]
```diff
@@ -4,3 +4,16 @@
   description: "A vending machine containing Security equipment. A label reads \"SECURITY PERSONNEL ONLY\"."
```
Comment (brndd) [inline Resources/Prototypes/Catalog/VendingMachines/seeds.yml:4]: guess botanists aren't allowed to use it then

File: Resources/Prototypes/Entities/Objects/Consumable/botany.yml [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+    - type: Sprite
+      sprite: Objects/Specific/Hydroponics/potato.rsi
+    - type: Produce
+      seed: potato
+
+- type: entity
+  name: tomato
+  parent: ProduceBase
+  id: FoodTomato
+  description: The space Irish starved to death after their potato crops died. Sadly they were unable to fish for space carp due to it being the queen's space. Bringing this up to any space IRA member will drive them insane with anger.
```
Comment (brndd) [inline Resources/Prototypes/Entities/Objects/Consumable/botany.yml:132]: wrong description

**Review by brndd — CHANGES_REQUESTED** [review]
There's an assload of "inconsistent braces style" errors in this from Rider because of all the braceless if statements but whatever.

File: Resources/Prototypes/Catalog/VendingMachines/seeds.yml (hunk 5 above)
Comment (Zumorica) [reply Resources/Prototypes/Catalog/VendingMachines/seeds.yml:4]: funny, this was like this before I even touched it, but a recent PR fixed it so it's fine now

File: Content.Shared/Atmos/Atmospherics.cs (hunk 3 above)
Comment (Zumorica) [reply Content.Shared/Atmos/Atmospherics.cs]: Fixed

File: Content.Shared/Interfaces/Chemistry/IPlantMetabolizable.cs (hunk 4 above)
Comment (Zumorica) [reply Content.Shared/Interfaces/Chemistry/IPlantMetabolizable.cs]: Changed to be proper now

File: Resources/Prototypes/Entities/Objects/Consumable/botany.yml (hunk 6 above)
Comment (Zumorica) [reply Resources/Prototypes/Entities/Objects/Consumable/botany.yml]: Copy-pasting is bad, kids. Fixed now

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs (hunk 1 above)
Comment (Zumorica) [reply Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:379]: Indeed, we delay it to the next tick

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs [hunk 7]
```diff
@@ -0,0 +1,850 @@
...
+        public void AdjustWater(float amount)
+        {
+            WaterLevel += amount;
+
+            // Water dilutes toxins.
+            if (amount > 0)
+            {
+                Toxins -= amount * 4f;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (Zumorica) [reply Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs:553]: I'd prefer if this stayed like this, to be honest, so I'll just make WaterLevel's setter private

File: Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs (hunk 2 above)
Comment (Zumorica) [reply Content.Server/GameObjects/Components/Botany/PlantHolderComponent.cs]: fixed

File: Content.Server/Chemistry/PlantMetabolism/AdjustHealth.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+namespace Content.Server.Chemistry.PlantMetabolism
+{
+    [UsedImplicitly]
+    public class AdjustHealth : AdjustAttribute
+    {
+        public override void Metabolize(IEntity plantHolder, float customPlantMetabolism = 1f)
+        {
+            if (plantHolder.Deleted || !plantHolder.TryGetComponent(out PlantHolderComponent? plantHolderComp)
+            || plantHolderComp.Seed == null || plantHolderComp.Dead || !CanMetabolize())
```
[the remark this answers is not in the document]
Comment (Zumorica) [reply Content.Server/Chemistry/PlantMetabolism/AdjustHealth.cs]: I made it prettier now, for most of the adjust attributes. Thanks!

**Pull request merged**